### PR TITLE
[WIP]  Use retry operation for all bulk retry types

### PR DESCRIPTION
**Pull request by janovesk**
Use includes to avoid queries and extra server roundtrips
Moved logic into RetryOperationManager

### CODE REVIEW

**Comment by janovesk** [comment]
Playing around with this @WilliamBZA. Trying to clean up and avoid the extra loads.

**Comment by janovesk** [comment]
Spike off from https://github.com/Particular/ServiceControl/pull/792

File: src/ServiceControl/Recoverability/Retries/RetryOperation.cs [hunk 1]
```diff
@@ -6,10 +6,24 @@ public class RetryOperation
         public string GroupId { get; set; }
         public int BatchesRemaining { get; set; }
         public int BatchesInOperation { get; set; }
+        public string RetryType { get; set; }
 
-        public static string MakeDocumentId(string messageUniqueId)
+
+        public static string MakeDocumentId(string identifier, RetriesGateway.RetryType retryType)
+        {
+            return $"RetryOperations/{retryType}/{identifier}";
```
Comment (WilliamBZA) [inline src/ServiceControl/Recoverability/Retries/RetryOperation.cs]: :+1:

**Comment by janovesk** [comment]
Good parts cherry picked into #792

**Pull request closed without merging**